### ISSUE TITLE
Release to Galactic/Rolling

### ISSUE THREAD

**Issue by emersonknapp** [stonier/ecl_core#112]
Howdy - noticed this is not in Galactic/Rolling ros2 distros and I would like to use it!

**Comment by fpasch** [comment]
I would also highly appreciate a release for galactic/rolling.

**Comment by ljaniec** [comment]
Same for `humble` now :)

**Comment by clalancette** [comment]
This has now been done for Foxy, Galactic, and Rolling, which is great.  Given our success on Rolling, I'm fairly confident that Humble would succeed.  @stonier , would you mind doing a release there?

**Comment by j-corvo** [comment]
Hello all,

Do you support ROS2 Jazzy? I would like to migrate my ROS1 robot to ROS2 Jazzy and I use this repository.

Thank you!

**Comment by emersonknapp** [comment]
Not involved in maintenance of this repo but I understand context on ROS builds, I'll walk through my brief investigation:

1. Having gone into Rolling, I believe these packages now automatically get included in new distributions in the branching process. However, that doesn't mean that the builds will automatically succeed.
2. You can see it in Humble/Jazzy in the ROS index at https://index.ros.org/p/ecl_core/#jazzy
3. Looking in the apt repo (http://packages.ros.org/ros2/ubuntu/dists/noble/main/binary-amd64/Packages) I am finding many of these `ecl` packages in Jazzy but not all of them, namely `ecl-core` which is the toplevel package that includes the rest.
4. The build for it is at https://build.ros2.org/job/Jbin_uN64__ecl_core__ubuntu_noble_amd64__binary/ which shows that a few of its dependency packages didn't build. As far as I can tell the weak link that's blocking the rest of the chain is `ecl_time_lite` https://build.ros2.org/job/Jbin_uN64__ecl_time_lite__ubuntu_noble_amd64__binary/

```
23:18:48 In file included from /tmp/binarydeb/ros-jazzy-ecl-time-lite-1.2.0/src/lib/../../include/ecl/time_lite/errors.hpp:21,
23:18:48                  from /tmp/binarydeb/ros-jazzy-ecl-time-lite-1.2.0/src/lib/../../include/ecl/time_lite/functions_rt.hpp:32,
23:18:48                  from /tmp/binarydeb/ros-jazzy-ecl-time-lite-1.2.0/src/lib/functions_rt.cpp:12:
23:18:48 /opt/ros/jazzy/include/ecl/errors/handlers.hpp:73:22: error: ‘virtual void ecl::Error::operator=(const ecl::ErrorFlag&)’ was hidden [-Werror=overloaded-virtual=]
23:18:48    73 |         virtual void operator=(const ErrorFlag &error) { error_flag = error; }
23:18:48       |                      ^~~~~~~~
23:18:48 /tmp/binarydeb/ros-jazzy-ecl-time-lite-1.2.0/src/lib/../../include/ecl/time_lite/errors.hpp:34:29: note:   by ‘ecl::TimeError::operator=’
23:18:48    34 | class ecl_time_lite_PUBLIC  TimeError : public Error {
23:18:48       |                             ^~~~~~~~~
23:18:48 cc1plus: all warnings being treated as errors
```

So I think what's happening here is that the `TimeError::TimeError(const ErrorFlag&)` constructor is not marked as `explicit`, so allows for implicit assignment, shadowing `virtual void Error::operator=(const ErrorFlag& error)` operator in the parent class.

Don't know if this is a new warning in the newer compiler version on Noble or what, because it's same code (and not actually a functional problem in this case, but a useful warning generally speaking) - or if warnings-as-errors got added at some point.

**Comment by emersonknapp** [comment]
Yeah seems like it's some new behavior as of GCC 13 which comes with Noble - not sure _exactly_ what the difference is. But I see now that `ecl_build` itself sets `-Werror` - https://github.com/stonier/ecl_tools/blob/devel/ecl_build/cmake/ecl_cxx.cmake#L29

**Comment by emersonknapp** [comment]
The problem wasn't about the constructor, but actually about some subtle name lookup stuff. Digging in that far got me on the hunt, so 

https://github.com/stonier/ecl_lite/pull/43

Should fix it all and get the rest of the packages released into Jazzy.

I don't have write access though so I can't merge that or do the subsequent bloom release to get the update into the buildfarm.

Hands off now, cheers!

**Comment by emersonknapp** [comment]
I'm actually going to close this issue as completed to spec, the package made it into Galactic & Rolling, then Humble automatically.
